### PR TITLE
[SPARK-16024] [SQL] [TEST] Verify Column Comment for Data Source Tables

### PR DESCRIPTION
**Pull request by gatorsmile**
#### What changes were proposed in this pull request?

This PR is to improve test coverage. It verifies whether `Comment` of `Column` can be appropriate handled. 

The test cases verify the related parts in Parser, both SQL and DataFrameWriter interface, and both Hive Metastore catalog and In-memory catalog.
#### How was this patch tested?

N/A

### CODE REVIEW

**Comment by gatorsmile** [comment]
cc @cloud-fan Tried to find the hole, but it sounds like all the cases can pass. Please let me know if anything is missing. Thanks!

**Comment by cloud-fan** [comment]
oh sorry looks like l made a false alarm, and thanks for adding these tests!

LGTM, can you resolve the conflict? thanks!

**Comment by gatorsmile** [comment]
Thank you! : ) @cloud-fan

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcQuerySuite.scala [hunk 1]
```diff
@@ -462,4 +463,27 @@ class OrcQuerySuite extends QueryTest with BeforeAndAfterAll with OrcTest {
       }
     }
   }
+
+  test("column nullability and comment - write and then read") {
+    val schema = StructType(
+      StructField("cl1", IntegerType, nullable = false,
+        new MetadataBuilder().putString("comment", "test").build()) ::
```
Comment (cloud-fan) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcQuerySuite.scala]: hmmm, is this the official way to add column comment when create table using `DataFrameWriter`?

cc @yhuai  @hvanhovell

Comment (gatorsmile) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcQuerySuite.scala]: : ) It is a little bit hacky. Maybe we should add a new API for users to add comments?

Comment (hvanhovell) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcQuerySuite.scala]: Yeah I am afraid it is. I just grepped through the code base and there are a few places where we do this, for example:
https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala#L1434-L1438
https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala#L397-L404

+1 for adding a convenience method.

Comment (gatorsmile) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcQuerySuite.scala]: Do you want me to submit a new PR for this? Or add it into this?

Comment (cloud-fan) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcQuerySuite.scala]: you can open a new PR and move this test there.

Comment (gatorsmile) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcQuerySuite.scala]: Sure, let me remove it and submit a new PR soon. Thanks!

File: sql/core/src/test/scala/org/apache/spark/sql/test/DataFrameReaderWriterSuite.scala [hunk 2]
```diff
@@ -223,6 +223,31 @@ class DataFrameReaderWriterSuite extends QueryTest with SharedSQLContext with Be
     }
   }
 
+  test("column nullability and comment - write and then read") {
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/test/DataFrameReaderWriterSuite.scala]: also remove this test, let's focus on SQL CREATE TABLE in this PR.

Comment (gatorsmile) [inline sql/core/src/test/scala/org/apache/spark/sql/test/DataFrameReaderWriterSuite.scala]: Sure, let me change it. Thanks!

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcQuerySuite.scala (hunk 1 above)
Comment (gatorsmile) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcQuerySuite.scala]: @hvanhovell @cloud-fan The new methods are added in the PR: https://github.com/apache/spark/pull/13860 Could you please review it? Thanks!

**Comment by cloud-fan** [comment]
thanks, merging to master and 2.0!

**Pull request closed without merging**